### ISSUE TITLE
As a grantee, I want to see 5 years of data submission history

### ISSUE THREAD

**Issue by ADPennington** [raft-tech/TANF-app#2267]
**Description:**

As part of pilot v2.2, a couple of grantees have expressed interest in being able to track 5 years of data submissions via TDP. They noted that this is helpful for their auditors. 

Some considerations:

- TDRS (legacy system) doesnt have history prior to FY21. 
- OFA is interested in retaining history for at least 5 years (and the updated SORN, once approved, should support this) 

**Acceptance Criteria:** 

_As a grantee, I want to see 5 years of data submission history_

- [ ] Allow users to submit past 4 years
- [ ] Ensure no side effects from submissions of past quarters (i.e emails from past quarters, SFTP resubmission)
- [ ] Testing Checklist has been run and all tests pass
- [ ] README is updated, if necessary

**Tasks:**

- [ ] Change the data file year dropdown to display last 5 years in reverse order
- [ ] Upon backend data file object creation, check year/quarter compared to present year/quarter to prevent side-effects
- [ ] Run Testing Checklist and confirm all tests pass

**Notes:**
_Add additional useful information, such as related issues and functionality that isn't covered by this specific issue, and other considerations that will be helpful for anyone reading this_ 

- _Note 1_
- _Note 2_
- _Note 3_

**Supporting Documentation:**
_Please include any relevant log snippets/files/screen shots_

- _Doc 1_
- _Doc 2_

**Open Questions:**
_Please include any questions or decisions that must be made before beginning work or to confidently call this issue complete_ 

- Does this mean TDP should keep the files and error reports available to users for 5 years? 
- is it 5 fiscal years or 5 years after the original submission date? 
- after this period, where should the history, files, and error reports go? 
- how does this relate to being able to submit previous fiscal years of data? Separate ticket?

**Comment by robgendron** [comment]
Blocked by SORN - need specific ticket to reference.

**Comment by ADPennington** [comment]
> Blocked by SORN - need specific ticket to reference.

@robgendron #1972 is the ticket blocking this one.

**Comment by vlasse86** [comment]
Questions:
- What is the workflow for removing options to submit data for prior years?
- How does that impact what's available on the front end?
- What's happening to the data in the db?

**Comment by ADPennington** [comment]
> Questions:
> 
> * What is the workflow for removing options to submit data for prior years?
> * How does that impact what's available on the front end?
> * What's happening to the data in the db?

OFA's data team met today to discuss this ticket, and we decided to close this one and open a new documentation-focused ticket. Will focus on multiple considerations:

- how long should information in the system be retained?
- how long should information in the system be visible to STT users?
- how long should information in the system be available to OFA users?
- If the availability of information in the system is capped,
  - what is the cap based on (calendar date? fiscal period? something else?)
  - will information be archived?
  - will information be deleted?

Information includes any db data (including metadata) as well as files in S3



cc: @lfrohlich @vlasse86
